numba/core/analysis.py

```diff
@@ -608,6 +608,13 @@ def find_literally_calls(func_ir, argtypes):
                     first_loc.setdefault(argindex, assign.loc)
     # Signal the dispatcher to force literal typing
     for pos in marked_args:
-        if not isinstance(argtypes[pos], (types.Literal, types.InitialValue)):
+        query_arg = argtypes[pos]
+        do_raise = (isinstance(query_arg, types.InitialValue) and
+                    query_arg.initial_value is None)
+        if do_raise:
+            loc = first_loc[pos]
+            raise errors.ForceLiteralArg(marked_args, loc=loc)
```

Comment:
Are there tests exercising this branch? I can't seem to find it

Reply:
my test here would trigger it but I am not sure if this is useful to raise ForceLiteralArg:

```python
from numba import njit, literally

@njit
def bar(d):
    return literally(d)

@njit
def inner(d):
    return bar(d)

@njit
def foo(x):
    d = {2: x}
    return inner(d)


r = foo(123)
print(r)
```
The literal request will never be satisfied it seems.

Reply:
try e.g. `./runtests.py numba.tests.test_dictobject.TestTypedDictInitialValues.test_homogeneous_and_literal`
 ?

Reply:
That test would pass for me even if the branch is commented out, but it did fail `numba.tests.test_dictobject.TestTypedDictInitialValues.test_mutation_not_carried_single_function`.
But it seems that it doesn't need to be `ForceLiteralArg` exception.

Reply:
The `ForceLiteralArg` would likely be useful if `prefer_literal=True`